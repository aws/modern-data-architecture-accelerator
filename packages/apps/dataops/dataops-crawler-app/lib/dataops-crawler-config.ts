/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps } from '@aws-caef/app';
import { CrawlerDefinition } from '@aws-caef/dataops-crawler-l3-construct';
import { CaefDataOpsConfigParser, CaefDataOpsConfigContents } from '@aws-caef/dataops-shared';
import { Schema } from "ajv";
import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';


export interface GlueCrawlerConfigContents extends CaefDataOpsConfigContents {
    /**
     * Name of the Data Ops project. The crawler config will be autowired to use existing resources deployed by the project.
     */
    projectName: string
    /**
     * Map of crawler names to crawler definitions 
     */
    crawlers: { [ key: string ]: CrawlerDefinition }
}

export class GlueCrawlerConfigParser extends CaefDataOpsConfigParser<GlueCrawlerConfigContents> {
    public readonly crawlerConfigs: { [ key: string ]: CrawlerDefinition }

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )

        this.crawlerConfigs = this.configContents.crawlers

    }

}