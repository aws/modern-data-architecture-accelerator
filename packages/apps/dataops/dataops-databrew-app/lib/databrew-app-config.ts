/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps } from "@aws-caef/app";
import { Schema } from "ajv";
import { Stack } from "aws-cdk-lib";
import * as configSchema from './config-schema.json';
import { DataBrewJobProps, DatasetProps, RecipeProps } from '@aws-caef/dataops-databrew-l3-construct';
import { CaefDataOpsConfigParser, CaefDataOpsConfigContents } from '@aws-caef/dataops-shared';

export interface DataBrewConfigContents extends CaefDataOpsConfigContents {


    // Name of the Data-Ops project.
    readonly projectName: string

    // List of recipes to be created.
    readonly recipes?: {
        [ key: string ]: RecipeProps
    }

    // List of recipes to be created.
    readonly datasets?: {
        [ key: string ]: DatasetProps
    }

    // List of recipes to be created.
    readonly jobs?: {
        [ key: string ]: DataBrewJobProps
    }
}

export class DataBrewConfigParser extends CaefDataOpsConfigParser<DataBrewConfigContents> {

    public readonly jobs?: Record<string, DataBrewJobProps>
    public readonly recipes?: Record<string, RecipeProps>
    public readonly datasets?: Record<string, DatasetProps>

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.jobs = this.configContents.jobs
        this.recipes = this.configContents.recipes
        this.datasets = this.configContents.datasets
    }
}

