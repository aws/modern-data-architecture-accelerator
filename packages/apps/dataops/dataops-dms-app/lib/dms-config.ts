/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import {  CaefAppConfigParserProps } from '@aws-caef/app';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';
import { CaefDataOpsConfigContents, CaefDataOpsConfigParser } from '@aws-caef/dataops-shared';
import { DMSProps } from '@aws-caef/dataops-dms-l3-construct';


export interface DMSConfigContents extends CaefDataOpsConfigContents {
    readonly dms: DMSProps
}


export class DMSConfigParser extends CaefDataOpsConfigParser<DMSConfigContents> {
    public dms: DMSProps
    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.dms = this.configContents.dms
    }
}

