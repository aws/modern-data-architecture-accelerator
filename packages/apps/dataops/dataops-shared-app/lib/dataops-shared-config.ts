/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from "@aws-caef/app";
import { CaefConfigTransformer, ICaefConfigTransformer, ICaefConfigValueTransformer } from "@aws-caef/config";
import { ICaefResourceNaming } from "@aws-caef/naming";
import { Schema } from "ajv";
import { Stack } from "aws-cdk-lib";


export interface CaefDataOpsConfigContents extends CaefBaseConfigContents {
    readonly securityConfigurationName: string
    readonly projectName: string
    readonly projectBucket: string
    readonly projectTopicArn: string
    readonly deploymentRole: string
    readonly kmsArn: string
}

export class CaefDataOpsConfigParser<T extends CaefDataOpsConfigContents> extends CaefAppConfigParser<T> {
    public readonly securityConfigurationName: string
    public readonly projectName: string
    public readonly projectBucket: string
    public readonly projectTopicArn: string
    public readonly deploymentRole: string
    public readonly kmsArn: string

    constructor( stack: Stack, props: CaefAppConfigParserProps, configSchema: Schema ) {
        super( stack, props, configSchema, [ new ProjectConfigTransformer( props.naming ) ] )
        this.securityConfigurationName = this.configContents.securityConfigurationName
        this.projectName = this.configContents.projectName
        this.projectBucket = this.configContents.projectBucket
        this.projectTopicArn = this.configContents.projectTopicArn
        this.deploymentRole = this.configContents.deploymentRole
        this.kmsArn = this.configContents.kmsArn
    }

}

class ProjectConfigTransformer implements ICaefConfigTransformer {
    private readonly naming: ICaefResourceNaming;
    constructor( naming: ICaefResourceNaming ) {
        this.naming = naming
    }
    public transformConfig ( config: { [ key: string ]: any; } ): { [ key: string ]: any; } {
        const projectName = config[ 'projectName' ]
        const moddedConfig = config
        moddedConfig[ "securityConfigurationName" ] = moddedConfig[ "securityConfigurationName" ] ? moddedConfig[ "securityConfigurationName" ] : "project:securityConfiguration/default"
        moddedConfig[ "projectBucket" ] = moddedConfig[ "projectBucket" ] ? moddedConfig[ "projectBucket" ] : "project:projectBucket/default"
        moddedConfig[ "projectTopicArn" ] = moddedConfig[ "projectTopicArn" ] ? moddedConfig[ "projectTopicArn" ] : "project:projectTopicArn/default"
        moddedConfig[ "deploymentRole" ] = moddedConfig[ "deploymentRole" ] ? moddedConfig[ "deploymentRole" ] : "project:deploymentRole/default"
        moddedConfig[ "kmsArn" ] = moddedConfig[ "kmsArn" ] ? moddedConfig[ "kmsArn" ] : "project:kmsArn/default"

        const projectConfigValTransformer = new ProjectConfigValueTransformer( projectName, this.naming )
        return new CaefConfigTransformer( projectConfigValTransformer ).transformConfig( moddedConfig )
    }
}

class ProjectConfigValueTransformer implements ICaefConfigValueTransformer {
    private readonly projectName: string;
    private readonly naming: ICaefResourceNaming;
    constructor( projectName: string, naming: ICaefResourceNaming ) {
        this.projectName = projectName
        this.naming = naming
    }
    public transformValue ( value: string ): string {
        if ( value.startsWith( "project:" ) ) {
            return "ssm:" + this.naming.ssmPath( `${ this.projectName }/${ value.split( ":" )[ 1 ] }`, false, false )
        } else {
            return value
        }
    }
}

