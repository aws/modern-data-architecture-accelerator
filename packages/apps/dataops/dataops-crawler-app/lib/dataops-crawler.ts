/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { GlueCrawlerL3Construct, GlueCrawlerL3ConstructProps } from '@aws-mdaa/dataops-crawler-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { GlueCrawlerConfigParser } from './dataops-crawler-config';

export class GlueCrawlerCDKApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }
  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const appConfig = new GlueCrawlerConfigParser(stack, parserProps);
    const constructProps: GlueCrawlerL3ConstructProps = {
      ...{
        crawlerConfigs: appConfig.crawlerConfigs,
        securityConfigurationName: appConfig.securityConfigurationName,
        projectName: appConfig.projectName,
        notificationTopicArn: appConfig.projectTopicArn,
      },
      ...l3ConstructProps,
    };
    new GlueCrawlerL3Construct(stack, 'construct', constructProps);
    return [stack];
  }
}
