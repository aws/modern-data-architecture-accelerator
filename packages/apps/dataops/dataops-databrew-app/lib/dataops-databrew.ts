/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps, CaefCdkApp } from '@aws-caef/app';
import { AppProps, Stack } from 'aws-cdk-lib';
import { DataBrewConfigParser } from './databrew-app-config';
import { CaefL3ConstructProps } from '@aws-caef/l3-construct';
import { DataBrewL3Construct, DataBrewL3ConstructProps } from '@aws-caef/dataops-databrew-l3-construct'

export class DataBrewApp extends CaefCdkApp {
    constructor( props: AppProps = {} ) {
        super( "dataops-databrew", props )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: CaefL3ConstructProps, parserProps: CaefAppConfigParserProps ) {
        const appConfig = new DataBrewConfigParser( stack, parserProps )
        const constructProps: DataBrewL3ConstructProps = {
            ...{
                datasets: appConfig.datasets,
                recipes: appConfig.recipes,
                projectName: appConfig.projectName,
                jobs: appConfig.jobs
            }, ...l3ConstructProps
        }
        new DataBrewL3Construct( stack, "jobs", constructProps );
        return [ stack ]
    }
}