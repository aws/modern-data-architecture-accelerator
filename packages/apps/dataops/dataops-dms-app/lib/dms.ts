/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { DMSL3Construct, DMSL3ConstructProps } from '@aws-mdaa/dataops-dms-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { DMSConfigParser } from './dms-config';

export class DMSCDKApp extends MdaaCdkApp {
    constructor( props: AppProps = {} ) {
        super( props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`) )
    }

    protected subGenerateResources ( stack: Stack, l3ConstructProps: MdaaL3ConstructProps, parserProps: MdaaAppConfigParserProps ) {
        const appConfig = new DMSConfigParser( stack, parserProps )

        const constructProps: DMSL3ConstructProps = {
            ...appConfig,
            ...l3ConstructProps
        }
        new DMSL3Construct( stack, "dms", constructProps );
        return [ stack ]
    }
}