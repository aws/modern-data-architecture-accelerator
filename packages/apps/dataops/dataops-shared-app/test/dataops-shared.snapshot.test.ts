/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { snapShotTest, snapShotTestApp, Create } from '@aws-mdaa/testing';
import { MdaaDataOpsConfigParser, MdaaDataOpsConfigContents } from '../lib/dataops-shared-config';
import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { StringParameter } from 'aws-cdk-lib/aws-ssm';
import { Schema } from 'ajv';
import * as path from 'path';

interface TestDataOpsConfigContents extends MdaaDataOpsConfigContents {
  readonly testParameter?: string;
}

class TestDataOpsConfigParser extends MdaaDataOpsConfigParser<TestDataOpsConfigContents> {
  public readonly testParameter?: string;

  constructor(stack: Stack, props: MdaaAppConfigParserProps) {
    super(stack, props, {} as Schema);
    this.testParameter = this.configContents.testParameter;
  }
}

class TestDataOpsSharedApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }

  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const configParser = new TestDataOpsConfigParser(stack, parserProps);

    new StringParameter(stack, 'SecurityConfigParam', {
      parameterName: l3ConstructProps.naming.ssmPath('security-config-name', true, false),
      stringValue: configParser.securityConfigurationName,
      description: 'DataOps security configuration name',
    });

    new StringParameter(stack, 'ProjectNameParam', {
      parameterName: l3ConstructProps.naming.ssmPath('project-name', true, false),
      stringValue: configParser.projectName,
      description: 'DataOps project name',
    });

    new StringParameter(stack, 'ProjectBucketParam', {
      parameterName: l3ConstructProps.naming.ssmPath('project-bucket', true, false),
      stringValue: configParser.projectBucket,
      description: 'DataOps project bucket reference',
    });

    new StringParameter(stack, 'ProjectTopicParam', {
      parameterName: l3ConstructProps.naming.ssmPath('project-topic-arn', true, false),
      stringValue: configParser.projectTopicArn,
      description: 'DataOps project topic ARN reference',
    });

    new StringParameter(stack, 'DeploymentRoleParam', {
      parameterName: l3ConstructProps.naming.ssmPath('deployment-role', true, false),
      stringValue: configParser.deploymentRole,
      description: 'DataOps deployment role reference',
    });

    new StringParameter(stack, 'KmsArnParam', {
      parameterName: l3ConstructProps.naming.ssmPath('kms-arn', true, false),
      stringValue: configParser.kmsArn,
      description: 'DataOps KMS ARN reference',
    });

    if (configParser.testParameter) {
      new StringParameter(stack, 'TestParam', {
        parameterName: l3ConstructProps.naming.ssmPath('test-parameter', true, false),
        stringValue: configParser.testParameter,
        description: 'Test parameter for DataOps shared config',
      });
    }

    return [stack];
  }
}
snapShotTest(
  'DataOps Shared Config',
  Create.stackProvider(
    'DataOpsSharedStackProject',
    (_, context) => {
      const moduleApp = new TestDataOpsSharedApp({
        context: {
          ...context,
          module_configs: path.join(__dirname, 'test-config.yaml'),
        },
      });
      return moduleApp.generateStack();
    },
    {
      module_name: 'test-dataops-shared-project',
      org: 'analytics-org',
      env: 'prod',
      domain: 'data-platform',
    },
  ),
);

snapShotTestApp(
  'DataOps Shared App',
  Create.appProvider(
    context => {
      const moduleApp = new TestDataOpsSharedApp({
        context: {
          ...context,
          module_configs: path.join(__dirname, 'test-config.yaml'),
        },
      });
      moduleApp.generateStack();
      return moduleApp;
    },
    {
      module_name: 'test-dataops-shared-app',
      org: 'test-org',
      env: 'test-env',
      domain: 'test-domain',
    },
  ),
);
