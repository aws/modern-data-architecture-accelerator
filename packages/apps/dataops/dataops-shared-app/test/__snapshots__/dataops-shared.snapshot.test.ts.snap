// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataOps Shared App App Snapshot Test 1`] = `
{
  "test-org-test-env-test-domain-test-dataops-shared-app": {
    "Description": "(SO0320-dataops-shared) Modern Data Architecture Accelerator. Version 1.1.0",
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomainanalyticsmlpipelinedeploymentRolemlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/analytics-ml-pipeline/deploymentRole/ml-pipeline",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomainanalyticsmlpipelinekmsArnanalyticsencryptionC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/analytics-ml-pipeline/kmsArn/analytics-encryption",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomainanalyticsmlpipelineprojectBucketanalyticsdataC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/analytics-ml-pipeline/projectBucket/analytics-data",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomainanalyticsmlpipelineprojectTopicArnmlnotificationsC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/analytics-ml-pipeline/projectTopicArn/ml-notifications",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomainanalyticsmlpipelinesecurityConfigurationmlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/analytics-ml-pipeline/securityConfiguration/ml-pipeline",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
    },
    "Resources": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps deployment role reference",
          "Name": "/test-org/test-domain/test-dataops-shared-app/deployment-role",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "SsmParameterValuetestorgtestdomainanalyticsmlpipelinedeploymentRolemlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps KMS ARN reference",
          "Name": "/test-org/test-domain/test-dataops-shared-app/kms-arn",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "SsmParameterValuetestorgtestdomainanalyticsmlpipelinekmsArnanalyticsencryptionC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps project bucket reference",
          "Name": "/test-org/test-domain/test-dataops-shared-app/project-bucket",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "SsmParameterValuetestorgtestdomainanalyticsmlpipelineprojectBucketanalyticsdataC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps project name",
          "Name": "/test-org/test-domain/test-dataops-shared-app/project-name",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "analytics-ml-pipeline",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps project topic ARN reference",
          "Name": "/test-org/test-domain/test-dataops-shared-app/project-topic-arn",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "SsmParameterValuetestorgtestdomainanalyticsmlpipelineprojectTopicArnmlnotificationsC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "DataOps security configuration name",
          "Name": "/test-org/test-domain/test-dataops-shared-app/security-config-name",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Ref": "SsmParameterValuetestorgtestdomainanalyticsmlpipelinesecurityConfigurationmlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Stack description parameter to update on version changes",
          "Name": "/test-org/test-domain/test-dataops-shared-app/aws-solution",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "(SO0320-dataops-shared) Modern Data Architecture Accelerator. Version 1.1.0",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Test parameter for DataOps shared config",
          "Name": "/test-org/test-domain/test-dataops-shared-app/test-parameter",
          "Tags": {
            "mdaa_cdk_app": "dataops-shared",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-dataops-shared-app",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "project-specific-ml-pipeline-config",
        },
        "Type": "AWS::SSM::Parameter",
      },
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5",
                    ],
                    {
                      "Ref": "BootstrapVersion",
                    },
                  ],
                },
              ],
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          },
        ],
      },
    },
  },
}
`;

exports[`DataOps Shared Config Snapshot Test 1`] = `
{
  "Description": "(SO0320-dataops-shared) Modern Data Architecture Accelerator. Version 1.1.0",
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinedeploymentRolemlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/analytics-ml-pipeline/deploymentRole/ml-pipeline",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinekmsArnanalyticsencryptionC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/analytics-ml-pipeline/kmsArn/analytics-encryption",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelineprojectBucketanalyticsdataC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/analytics-ml-pipeline/projectBucket/analytics-data",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelineprojectTopicArnmlnotificationsC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/analytics-ml-pipeline/projectTopicArn/ml-notifications",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinesecurityConfigurationmlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/analytics-ml-pipeline/securityConfiguration/ml-pipeline",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueanalyticsorgdataplatformcaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/analytics-org/data-platform/caef-bootstrap/role-helper-service-token",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps deployment role reference",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/deployment-role",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinedeploymentRolemlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps KMS ARN reference",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/kms-arn",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinekmsArnanalyticsencryptionC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps project bucket reference",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/project-bucket",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelineprojectBucketanalyticsdataC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps project name",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/project-name",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": "analytics-ml-pipeline",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps project topic ARN reference",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/project-topic-arn",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelineprojectTopicArnmlnotificationsC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "DataOps security configuration name",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/security-config-name",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": {
          "Ref": "SsmParameterValueanalyticsorgdataplatformanalyticsmlpipelinesecurityConfigurationmlpipelineC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Stack description parameter to update on version changes",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/aws-solution",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": "(SO0320-dataops-shared) Modern Data Architecture Accelerator. Version 1.1.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Test parameter for DataOps shared config",
        "Name": "/analytics-org/data-platform/test-dataops-shared-project/test-parameter",
        "Tags": {
          "mdaa_cdk_app": "dataops-shared",
          "mdaa_domain": "data-platform",
          "mdaa_env": "prod",
          "mdaa_module_name": "test-dataops-shared-project",
          "mdaa_org": "analytics-org",
        },
        "Type": "String",
        "Value": "project-specific-ml-pipeline-config",
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
